src/thor/timedep_reverse.cc

```diff
@@ -215,15 +214,16 @@ inline bool TimeDepReverse::ExpandReverseInner(GraphReader& graphreader,
 
   Cost tc =
       costing_->TransitionCostReverse(meta.edge->localedgeidx(), nodeinfo, opp_edge, opp_pred_edge);
-  Cost newcost = pred.cost() + costing_->EdgeCost(opp_edge, t2, seconds_of_week);
+  auto edge_cost = costing_->EdgeCost(opp_edge, t2, seconds_of_week);
+  Cost newcost = pred.cost() + edge_cost;
   newcost.cost += tc.cost;
 
   // If this edge is a destination, subtract the partial/remainder cost
   // (cost from the dest. location to the end of the edge).
-  auto p = destinations_.find(meta.edge_id);
-  if (p != destinations_.end()) {
-    // Subtract partial cost and time
-    newcost -= p->second;
+  auto p = destinations_percent_along_.find(meta.edge_id);
+  if (p != destinations_percent_along_.end()) {
+    // Adapt cost to potentially not using the entire destination edge
+    newcost -= edge_cost * (1.0f - p->second);
```

Comment:
should this be `1 - partial` in the reverse direction? i remember confusing myself about this a bunch. since its the opposing edge i guess it should be right?

Reply:
I thought I had this unit-tested. Turns out the unittest only tested trivial paths. I added a unittest for non-trivial reverse search and it correctly pointed out that the above is wrong.

The new commit therefore updates this code to `newcost -= edge_cost * p->second;`